packages/api-server/src/serverManager.ts

```diff
@@ -82,13 +83,39 @@ export class ServerManager {
   }
 
   async restartApiServer() {
-    this.killApiServer()
+    await this.killApiServer()
     await this.startApiServer()
   }
 
-  killApiServer() {
-    this.httpServerProcess?.emit('exit')
-    this.httpServerProcess?.kill()
+  async killApiServer() {
+    if (!this.httpServerProcess) {
+      return
+    }
+
+    // Try to gracefully close the server
+    // If it doesn't close within 2 seconds, forcefully close it
+    await Promise.race([
+      new Promise<void>((resolve) => {
+        console.log(
+          chalk.yellow(
+            'API server did not exit within 2 seconds, forcefully closing it.',
```

Comment:
**Nitpick**:  This message seems inaccurate? This is the first attempt at closing the  API server, so maybe we should change it to something like "Shutting down API server."

Reply:
Whoops, that's a left-out-test-log. Since I could not repro the delated kill, I wanted to see what "yellow" looked like, so I moved up to the "normal" case. 

Thanks for catching it!